dlt/extract/decorators.py

```diff
@@ -164,6 +168,10 @@ def _wrap(*args: Any, **kwargs: Any) -> DltSource:
                     if inspect.isgenerator(rv):
                         rv = list(rv)
 
+            # prepare schema
```

Comment:
handle it in the same way as `root_key` and `max_table_nesting`. as a property of DltSource that does the magic inside